include/boost/math/special_functions/gamma.hpp

```diff
@@ -14,6 +14,22 @@
 #pragma once
 #endif
 
+#ifdef __CUDACC_RTC__
+
+namespace boost {
+namespace math {
+
+template <typename T>
+__host__ __device__ T tgamma(T x)
```

Comment:
It's somewhat a matter of taste, but I would probably go with:

```
inline __host__ __device__ float tgamma(float x) { return ::tgammaf(x); }
inline __host__ __device__ double tgamma(double x) { return ::tgamma(x); }
```

Certainly it should be inline at least.  I would also add overloads that accept a Policy (and which ignore it), so that when other code calls tgamma it has something to forward to.

Reply:
I think the template functions are automatically inline, but I agree about having separate float and double versions, if that’s what you mean here. Usually you see things like having minimax polynomial expansions which are different for different precision levels. (I know you know this @jzmaddock because I’ve seen this done in the boost math codebase, just pointing it out for everyone else.)